src/bionics.cpp

```diff
@@ -67,10 +69,141 @@ const efftype_id effect_visuals( "visuals" );
+        for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+            mvwprintw( win, 3, pos[i], titles[i].c_str() );
+        }
+    }
+    wrefresh( win );
+}
+
+void draw_header( WINDOW *win, const size_t tab_index, const std::string power_string,
+                  const std::string help_key )
+{
+    const std::array<std::string, 3> titles = {{
+        string_format( _( "Body Parts <color_yellow><< %s >></color>" ), bodyparts[tab_index].c_str() ),
+        power_string,
+        string_format( _( "Press '<color_yellow>%s</color>' for help" ), help_key.c_str() )
+    }};
+    // left / center / right aligned elements
+    const std::array<int, 3> pos = {{
+        0,
+        ( getmaxx( win ) - utf8_width( titles[1] ) ) / 2 + 1,
+        ( getmaxx( win ) - utf8_width( titles[2], true ) - 1 )
+    }};
+
+    werase( win );
+    for( size_t i = 0; i < std::min( titles.size(), pos.size() ); ++i ) {
+        fold_and_print( win, 0, pos[i], getmaxx( win ) - 3, c_white, titles[i].c_str() );
+    }
+    wrefresh( win );
+}
+
+const char *power_description( std::string const &id )
+{
+    std::ostringstream power_desc;
+    bool has_previous_text = false;
+    if( bionics[id].power_over_time > 0 && bionics[id].charge_time > 0 ) {
+        power_desc << ( bionics[id].charge_time == 1 ?
+                        string_format( _("%d PU / turn" ), bionics[id].power_over_time ) :
+                        string_format( _("%d PU / %d turns" ), bionics[id].power_over_time,
+                                       bionics[id].charge_time ) );
+        has_previous_text = true;
+    }
+    if( bionics[id].power_activate > 0 && bionics[id].charge_time == 0 ) {
+        if( has_previous_text ) {
+            power_desc << ", ";
+        }
+        power_desc << string_format( _( "%d PU / activation" ),
+                                     bionics[id].power_activate );
+        has_previous_text = true;
+    }
+    if( bionics[id].power_deactivate > 0 && bionics[id].charge_time == 0 ) {
+        if( has_previous_text ) {
+            power_desc << ", ";
+        }
+        power_desc << string_format( _( "%d PU / deactivation" ),
+                                     bionics[id].power_deactivate );
+    }
+    return power_desc.str().c_str();
+}
+
+bool compare_cbm_names( cbm_pair val1, cbm_pair val2 )
+{
+    if( val1.second == INT_MAX ){
+        return true;
+    } else if( val2.second == INT_MAX ) {
+        return false;
+    }
+    std::string name1 = bionics[g->u.my_bionics[val1.second].id].name;
+    std::string name2 = bionics[g->u.my_bionics[val2.second].id].name;
+    return( name1.compare( name2 ) <= 0 );
+}
+
+std::vector<cbm_pair> define_content( body_part bp )
+{
+    std::vector<cbm_pair> content;
+    content.push_back( std::make_pair( bp , INT_MAX ) );
+    for( size_t i = 0; i < g->u.my_bionics.size(); ++i ) {
+        if( g->u.my_bionics[i].occupied_bp == bp ) {
```

Comment:
Since you invoke `define_content` from the `player` class, it's probably better not to tie it to `g->u`. You could either pass `player ` as the first parameter or transform `define_content` to `player`'s method.